### PR TITLE
Design document for line card hot swapping support in SONiC.

### PR DESCRIPTION
**Pull request by inspurSDN**
Design document for line card hot swapping support in SONiC.

### CODE REVIEW

File: doc/linecard_hotswap/LineCard_hotSwap_HLD.md [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+For stability, it is not allowed to reboot the device after hot-swapping hardware components. The swappable design is very important for data center to minimize the impact of failures on certain hardware components on the switch. 
+This HLD provides a generic software mechanism of line card hot swapping, which includes the architecture, functional flow and CLI command.
+
+## 2 Requirements<a name="requirements"></a>
+
+### 2.1 Functional Requirements<a name="functionality"></a>
+
+**This feature will support the following functionality:**
+1.	1 chassis consists of swappable line cards.
+2.	Every line card with the same card type, 32*100G(Different card type will be implemented in the near future.)
```
Comment (lguohan) [inline doc/linecard_hotswap/LineCard_hotSwap_HLD.md]: please have a clear definition of line card.

Comment (lguohan) [reply doc/linecard_hotswap/LineCard_hotSwap_HLD.md]: what is the requirement of phy driver, does it requires firmware download for the phy driver.

Comment (inspurSDN) [reply doc/linecard_hotswap/LineCard_hotSwap_HLD.md]: OK, we will define what is included in the line card.

**Comment by lguohan** [comment]
please refer to this doc.

https://github.com/Azure/SONiC/pull/538

File: doc/linecard_hotswap/LineCard_hotSwap_HLD.md (hunk 1 above)
Comment (inspurSDN) [reply doc/linecard_hotswap/LineCard_hotSwap_HLD.md]: We update the definition of line card. For the requirement of phy driver, we can based on  gearbox design to use a profile while PHY initializes. I will integrate our design with #538

File: doc/linecard_hotswap/LineCard_hotSwap_HLD.md [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+
+OrchAgent will register an event callback function to publish the line card hot swapping events to DB and PortsOrch will be responsible for processing the line card events.
+
+### 3.3 Syncd Changes<a name="syncd-changes"></a>
+
+A new table, Port State Table, is created in syncd to record port present status. When user configure a port via SAI, syncd will check the port state in Port State Table. If the port is not present, the configuration will not be applied on that port and return SUCCESS.
+
+
+## 3.4 DB Changes<a name="db-changes"></a>
+
```
Comment (stephenxs) [inline doc/linecard_hotswap/LineCard_hotSwap_HLD.md:117]: How can I get the line-card no. from a port? Is there any table in config db or state db that indicates that? Or it should be extracted from port name?
Thanks.

Comment (inspurSDN) [reply doc/linecard_hotswap/LineCard_hotSwap_HLD.md:117]: Actually, We have a new design which based on Gearbox. In the new design, we may get the line card no. from the PHY profile.

Comment (stephenxs) [reply doc/linecard_hotswap/LineCard_hotSwap_HLD.md:117]: where can I find it?

Comment (inspurSDN) [reply doc/linecard_hotswap/LineCard_hotSwap_HLD.md:117]: @stephenxs  Please refer the new commit b9e088a

Comment (stephenxs) [reply doc/linecard_hotswap/LineCard_hotSwap_HLD.md:117]: Thanks. Will check it.

**Pull request closed without merging**